Matrix/joint_matrix_tensorcore_double.cpp fails on CUDA

https://github.com/intel/llvm/commit/9abe2016b4177f02903e88fcc973236d717ddc5c by @capatober broke Matrix/joint_matrix_tensorcore_double.cpp on Linux CUDA. The patch does not modify the matrix functionality, but put other functions to experimental namespace. 

I suppose this exposed a bug in the matrix feature implementation.

<details>
<summary>Full log</summary>
From http://icl-jenkins.sc.intel.com:8080/blue/organizations/jenkins/LLVM-Test-Suite-CI-TMP%2FLLVM-Test-Suite-CUDA/detail/LLVM-Test-Suite-CUDA/1212/pipeline

[2021-12-23T06:53:06.703Z] FAIL: SYCL :: Matrix/joint_matrix_tensorcore_double.cpp (505 of 688)

[2021-12-23T06:53:06.703Z] ******************** TEST 'SYCL :: Matrix/joint_matrix_tensorcore_double.cpp' FAILED ********************

[2021-12-23T06:53:06.703Z] Script:

[2021-12-23T06:53:06.703Z] --

[2021-12-23T06:53:06.703Z] : 'RUN: at line 3';    /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/clang++      -fsycl -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --cuda-gpu-arch=sm_80 -DSYCL_EXT_ONEAPI_MATRIX=3  /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm-test-suite/SYCL/Matrix/joint_matrix_tensorcore_double.cpp -o /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/build/SYCL/Matrix/Output/joint_matrix_tensorcore_double.cpp.tmp.out

[2021-12-23T06:53:06.703Z] --

[2021-12-23T06:53:06.703Z] Exit Code: 1

[2021-12-23T06:53:06.703Z] 

[2021-12-23T06:53:06.703Z] Command Output (stdout):

[2021-12-23T06:53:06.703Z] --

[2021-12-23T06:53:06.703Z] $ ":" "RUN: at line 3"

[2021-12-23T06:53:06.703Z] note: command had no output on stdout or stderr

[2021-12-23T06:53:06.703Z] $ "/localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/clang++" "-fsycl" "-fsycl-targets=nvptx64-nvidia-cuda" "-Xsycl-target-backend" "--cuda-gpu-arch=sm_80" "-DSYCL_EXT_ONEAPI_MATRIX=3" "/localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm-test-suite/SYCL/Matrix/joint_matrix_tensorcore_double.cpp" "-o" "/localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/build/SYCL/Matrix/Output/joint_matrix_tensorcore_double.cpp.tmp.out"

[2021-12-23T06:53:06.703Z] # command stderr:

[2021-12-23T06:53:06.703Z] In file included from /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm-test-suite/SYCL/Matrix/joint_matrix_tensorcore_double.cpp:10:

[2021-12-23T06:53:06.703Z] In file included from /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/../include/sycl/CL/sycl.hpp:66:

[2021-12-23T06:53:06.703Z] In file included from /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/../include/sycl/ext/oneapi/matrix/matrix.hpp:29:

[2021-12-23T06:53:06.703Z] /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/../include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp:228:3: error: no template named 'joint_matrix_load_impl' in namespace 'sycl::ext::oneapi::experimental::detail'; did you mean '::sycl::ext::oneapi::detail::joint_matrix_load_impl'?

[2021-12-23T06:53:06.703Z]   detail::joint_matrix_load_impl<T, MT, NumRows, NumCols, Layout, Space>{}.load(

[2021-12-23T06:53:06.703Z]   ^~~~~~~~

[2021-12-23T06:53:06.703Z] /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/../include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp:59:8: note: '::sycl::ext::oneapi::detail::joint_matrix_load_impl' declared here

[2021-12-23T06:53:06.703Z] struct joint_matrix_load_impl {

[2021-12-23T06:53:06.703Z]        ^

[2021-12-23T06:53:06.703Z] /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/../include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp:238:3: error: no template named 'joint_matrix_store_impl' in namespace 'sycl::ext::oneapi::experimental::detail'; did you mean '::sycl::ext::oneapi::detail::joint_matrix_store_impl'?

[2021-12-23T06:53:06.703Z]   detail::joint_matrix_store_impl<T, NumRows, NumCols, Layout, Space>{}.store(

[2021-12-23T06:53:06.703Z]   ^~~~~~~~

[2021-12-23T06:53:06.703Z] /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/../include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp:127:8: note: '::sycl::ext::oneapi::detail::joint_matrix_store_impl' declared here

[2021-12-23T06:53:06.703Z] struct joint_matrix_store_impl {

[2021-12-23T06:53:06.703Z]        ^

[2021-12-23T06:53:06.703Z] /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/../include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp:250:10: error: no template named 'joint_matrix_mad_impl' in namespace 'sycl::ext::oneapi::experimental::detail'; did you mean '::sycl::ext::oneapi::detail::joint_matrix_mad_impl'?

[2021-12-23T06:53:06.703Z]   return detail::joint_matrix_mad_impl<T1, T2, M, K, N, LayoutA, LayoutB,

[2021-12-23T06:53:06.703Z]          ^~~~~~~~

[2021-12-23T06:53:06.703Z] /localdisk2/iusers/sys_bbsycl/llvm-test-suite_Jenkins/workspace/LLVM-Test-Suite-CI-TMP/LLVM-Test-Suite-CUDA/llvm.obj/bin/../include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp:154:8: note: '::sycl::ext::oneapi::detail::joint_matrix_mad_impl' declared here

[2021-12-23T06:53:06.703Z] struct joint_matrix_mad_impl {

[2021-12-23T06:53:06.703Z]        ^

[2021-12-23T06:53:06.703Z] 3 errors generated.

[2021-12-23T06:53:06.703Z] 

[2021-12-23T06:53:06.703Z] error: command failed with exit status: 1

[2021-12-23T06:53:06.703Z] 

[2021-12-23T06:53:06.703Z] --

[2021-12-23T06:53:06.703Z] 

[2021-12-23T06:53:06.703Z] ********************
</details>

--- a/SYCL/Matrix/joint_matrix_tensorcore_double.cpp
+++ b/SYCL/Matrix/joint_matrix_tensorcore_double.cpp
@@ -1,5 +1,4 @@
-// The test is disabled to devil's issue #666
-// REQUIRES: gpu, cuda, TEMPORARY_DISABLED
+// REQUIRES: gpu, cuda
 
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -Xsycl-target-backend --cuda-gpu-arch=sm_80 -DSYCL_EXT_ONEAPI_MATRIX=3  %s -o %t.out
 //

